### PR TITLE
DBZ-1296 Describe SQL Server connector and events

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-1296

### CODE REVIEW

File: docs/connectors/sqlserver.asciidoc [hunk 1]
```diff
@@ -96,7 +96,451 @@ We welcome any feedback from a user to try the plug-in with database in managed
...
+                "type": "int32",
+                "optional": false,
+                "field": "id"
+            }
+        ],
+        "optional": false,
+        "name": "server1.dbo.customers.Key"
+    },
+    "payload": {
+        "ID": 1004
```
Comment (gunnarmorling) [inline docs/connectors/sqlserver.asciidoc]:
```suggestion
        "id": 1004
```

File: docs/connectors/sqlserver.asciidoc [hunk 2]
```diff
@@ -96,7 +96,451 @@ We welcome any feedback from a user to try the plug-in with database in managed
...
+[[change-event-values]]
+==== Change Event Values
+
+Like the message key, the value of a change event message has a _schema_ section and _payload_ section.
+The payload section of every change event value produced by the SQL Server connector has an _envelope_ structure with the following fields:
+
+* `op` is a mandatory field that contains a string value describing the type of operation. Values for the SQL Server connector are `c` for create (or insert), `u` for update, `d` for delete, and `r` for read (in the case of a snapshot).
+* `before` is an optional field that if present contains the state of the row _before_ the event occurred. The structure will be described by the `server1.dbo.customers.Value` Kafka Connect schema, which the `server1` connector uses for all rows in the `dbo.customers` table.
+
+* `after` is an optional field that if present contains the state of the row _after_ the event occurred. The structure is describe by the same `server1.dbo.customers.Value` Kafka Connect schema used in `before`.
```
Comment (gunnarmorling) [inline docs/connectors/sqlserver.asciidoc]:
```suggestion
* `after` is an optional field that if present contains the state of the row _after_ the event occurred. The structure is described by the same `server1.dbo.customers.Value` Kafka Connect schema used in `before`.
```

File: docs/connectors/sqlserver.asciidoc [hunk 3]
```diff
@@ -96,7 +96,451 @@ We welcome any feedback from a user to try the plug-in with database in managed
...
+The SQL Server connector's events are designed to work with https://cwiki.apache.org/confluence/display/KAFKA/Log+Compaction[Kafka log compaction],
+which allows for the removal of some older messages as long as at least the most recent message for every key is kept.
+This allows Kafka to reclaim storage space while ensuring the topic contains a complete dataset and can be used for reloading key-based state.
+
+[[tombstone-events]]
+When a row is deleted, the _delete_ event value listed above still works with log compaction, since Kafka can still remove all earlier messages with that same key.
+But only if the message value is `null` will Kafka know that it can remove _all messages_ with that same key.
+To make this possible, Debezium's SQL Server connector always follows the _delete_ event with a special _tombstone_ event that has the same key but `null` value.
+
+[NOTE]
```
Comment (gunnarmorling) [inline docs/connectors/sqlserver.asciidoc]: https://issues.apache.org/jira/browse/KAFKA-3832 is resolved in Kafka 2.0.2, so this note can be removed. I'll pushe a commit for this.

**Comment by gunnarmorling** [comment]
@jpechane Great work! I've pushed two commits. Can you apply that last pending comment of mine and then merge everything? Thanks!

**Comment by jpechane** [comment]
@gunnarmorling Applied, thanks for the cleanup!

**Pull request merged**